labelbox/schema/project.py

```diff
@@ -608,22 +608,31 @@ def create_batch(self,
 
         self._wait_until_data_rows_are_processed(
             dr_ids, self._wait_processing_max_seconds)
-        method = 'createBatchV2'
-        query_str = """mutation %sPyApi($projectId: ID!, $batchInput: CreateBatchInput!) {
-              project(where: {id: $projectId}) {
-                %s(input: $batchInput) {
-                    batch {
-                        %s
-                    }
-                    failedDataRowIds
-                }
-              }
-            }
-        """ % (method, method, query.results_query_part(Entity.Batch))
 
         if consensus_settings:
             consensus_settings = ConsensusSettings(**consensus_settings).dict(
                 by_alias=True)
+
+        if len(dr_ids) >= 10_000:
+            return self._create_batch_async(name, dr_ids, priority,
+                                            consensus_settings)
+        else:
+            return self._create_batch_sync(name, dr_ids, priority,
```

Comment:
Do we check that the sync create doesn't exceed 10k data rows on the backend? That might be a good safety check if we don't already. Especially since they can just call project._create_batch_sync() with more than 10k data rows.

Reply:
Yes, it is checked on the backend for the synchronous endpoints. There's a feature flag that controls the maximum